src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+  render(<Textarea required label="required" />);
+  expect(screen.getByText("required")).toHaveStyleRule("content", '"*"', {
+    modifier: "::after",
+  });
+});
+
+test("should set the isOptional styling on both input labels when prop is true", () => {
+  render(<Textarea isOptional label="optional" />);
+  expect(screen.getByTestId("label-container")).toHaveStyleRule(
+    "content",
+    '"(optional)"',
+    {
+      modifier: "::after",
+    }
+  );
+});
+
+describe("when rendered with new validations", () => {
+  const renderWithNewValidations = (props: TextareaProps) =>
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <Textarea {...props} />
+      </CarbonProvider>
+    );
+
+  it.each(["error", "warning"])(
+    'adds the id of the validation text to "aria-describedby" in the textarea element',
+    (validationType) => {
+      const mockId = "foo";
+      renderWithNewValidations({ id: mockId, [validationType]: "bar" });
+
+      expect(screen.getByText("bar")).toHaveAttribute(
+        "id",
+        `${mockId}-validation`
+      );
+      expect(screen.getByRole("textbox")).toHaveAttribute(
+        "aria-describedby",
+        `${mockId}-validation`
+      );
+    }
+  );
+
+  it("ignores the labelInline and related styling props", () => {
+    renderWithNewValidations({
+      labelInline: true,
+      label: "example label",
+      labelAlign: "left",
+      labelWidth: 100,
+      labelSpacing: 1,
+    });
+    const labelContainer = screen.getByTestId("label-container");
+    expect(labelContainer).toHaveStyle({ width: undefined });
+    expect(labelContainer).toHaveStyle({ justifyContent: undefined });
+    expect(labelContainer).toHaveStyle({ paddingLeft: undefined });
+    expect(labelContainer).toHaveStyle({ paddingRight: undefined });
+  });
+
+  it("renders the hint text with the correct styling when the labelHelp prop is passed", () => {
+    renderWithNewValidations({ labelHelp: "Example hint text" });
+    const hintText = screen.getByText("Example hint text");
+    expect(hintText).toBeInTheDocument();
+    expect(hintText).toHaveStyle({
+      fontSize: "14px",
+      marginTop: "0px",
+      marginBottom: "8px",
+    });
+    expect(hintText).toHaveStyleRule("color", "var(--colorsUtilityYin055)");
+  });
+});
+
+test("accepts ref as a ref object", () => {
+  const ref = { current: null };
+  render(<Textarea ref={ref} />);
+
+  expect(ref.current).toBe(screen.getByRole("textbox"));
+});
+
+test("accepts ref as a ref callback", () => {
+  const ref = jest.fn();
+  render(<Textarea ref={ref} />);
+
+  expect(ref).toHaveBeenCalledWith(screen.getByRole("textbox"));
+});
+
+test("sets ref to empty after unmount", () => {
+  const ref = { current: null };
+  const { unmount } = render(<Textarea ref={ref} />);
+
+  unmount();
+
+  expect(ref.current).toBe(null);
+});
+
+test("renders with the expected default border radius styling", () => {
```

Comment:
comment (non-blocking): I think we should remove any style tests that aren't needed for coverage that are already captured by chromatic/playwright. I'm happy for that to be done in a new ticket as these pass and aren't really hurting anything right now if we leave it etc

Reply:
In fact lets leave this for now and do it in a second pass as we'll have other test files that are doing the same etc